SPECS/rust/rust.spec

```diff
@@ -1,39 +1,41 @@
 # Prevent librustc_driver from inadvertently being listed as a requirement
 %global __requires_exclude ^librustc_driver-
+# Release date and version of stage 0 compiler can be found in "src/stage0.txt" inside the extracted "Source0".
+# Look for "date:" and "rustc:".
+%define release_date 2021-09-09
+%define stage0_version 1.55.0
 Summary:        Rust Programming Language
 Name:           rust
-Version:        1.47.0
-Release:        5%{?dist}
+Version:        1.56.1
+Release:        1%{?dist}
 License:        ASL 2.0 AND MIT
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          Applications/System
 URL:            https://www.rust-lang.org/
-Source0:        https://static.rust-lang.org/dist/rustc-%{version}-src.tar.xz
+Source0:        https://static.rust-lang.org/dist/rustc-%{version}-src.tar.gz
+# Note: the rust-%%{version}-cargo.tar.gz file contains a cache created by capturing the contents downloaded into $CARGO_HOME.
+# To update the cache run:
+#   [repo_root]/toolkit/scripts/build_cargo_cache.sh rustc-%%{version}-src.tar.gz
 Source1:        %{name}-%{version}-cargo.tar.gz
-Source2:        https://static.rust-lang.org/dist/2020-08-27/cargo-0.47.0-x86_64-unknown-linux-gnu.tar.gz
-Source3:        https://static.rust-lang.org/dist/2020-08-27/rustc-1.46.0-x86_64-unknown-linux-gnu.tar.gz
-Source4:        https://static.rust-lang.org/dist/2020-08-27/rust-std-1.46.0-x86_64-unknown-linux-gnu.tar.gz
-Source5:        https://static.rust-lang.org/dist/2020-08-27/cargo-0.47.0-aarch64-unknown-linux-gnu.tar.gz
-Source6:        https://static.rust-lang.org/dist/2020-08-27/rustc-1.46.0-aarch64-unknown-linux-gnu.tar.gz
-Source7:        https://static.rust-lang.org/dist/2020-08-27/rust-std-1.46.0-aarch64-unknown-linux-gnu.tar.gz
-Patch0:         CVE-2020-36317.patch
-Patch1:         CVE-2021-28875.patch
-Patch2:         CVE-2021-28877.patch
-Patch3:         CVE-2021-28876.patch
-Patch4:         CVE-2021-28879.patch
-Patch5:         CVE-2021-28878.patch
-Patch6:         CVE-2020-36323.patch
+Source2:        https://static.rust-lang.org/dist/%{release_date}/cargo-%{stage0_version}-x86_64-unknown-linux-gnu.tar.gz
+Source3:        https://static.rust-lang.org/dist/%{release_date}/rustc-%{stage0_version}-x86_64-unknown-linux-gnu.tar.gz
+Source4:        https://static.rust-lang.org/dist/%{release_date}/rust-std-%{stage0_version}-x86_64-unknown-linux-gnu.tar.gz
+Source5:        https://static.rust-lang.org/dist/%{release_date}/cargo-%{stage0_version}-aarch64-unknown-linux-gnu.tar.gz
+Source6:        https://static.rust-lang.org/dist/%{release_date}/rustc-%{stage0_version}-aarch64-unknown-linux-gnu.tar.gz
+Source7:        https://static.rust-lang.org/dist/%{release_date}/rust-std-%{stage0_version}-aarch64-unknown-linux-gnu.tar.gz
 
 BuildRequires:  binutils
 BuildRequires:  cmake
 BuildRequires:  curl-devel
 BuildRequires:  git
 BuildRequires:  glibc
+BuildRequires:  ninja-build
 BuildRequires:  python2
```

Comment:
Can we replace `python2` with `python3` here? Should be a drop-in replacement. 

Actually, looking at the [`x.py` script](https://github.com/rust-lang/rust/blob/1.56.1/x.py), it looks like python2 isn't used anyways when python3 is present.

Reply:
Good idea, done.